File: tournaments/differentValues/differentValues.py
def differentValues(a, d):
    result = -1
    
    for i in range(len(a)):
        for j in range(i + 1, len(a)):
            diff = abs(a[j] - a[i])
            if 0 <= d - diff and d - diff <= d - result:
                result = diff
    
    return result

File: tournaments/differentValues/differentValues.js
function differentValues(a, d) {
    const n = a.length;
    let result = -1;
    for (let i = 0; n > i; ++i)
        for (let j = i + 1; n > j; ++j)
            if (d >= Math.abs(a[i] - a[j])) result = Math.max(result, Math.abs(a[i] - a[j]));
    return result;
}
